sdk/src/trace/batch_span_processor.cc

```diff
@@ -184,37 +212,25 @@ void BatchSpanProcessor::Export(const bool was_force_flush_called)
     {
       exporter_->Export(
           nostd::span<std::unique_ptr<Recordable>>(spans_arr.data(), spans_arr.size()));
+      NotifyCompletion(notify_force_flush, synchronization_data_);
```

Comment:
I am not clear on the scenario when by the time sync Export completes and notifies the main thread about force flush completion, buffer gets updated with new recordables and then Export would happen again. Should the main thread wait for second export ? or its not considered as a part of force flush?

Reply:
I think `ForceFlush` means flush and wait all the data before the calling time. It may be used to ensure logical order.(To keep A happens or not happens before B, just like [memory_order](https://en.cppreference.com/w/c/atomic/memory_order))
I believe it do not concerns what will happend in the future.